### DISCUSSION TITLE
Observations found in Zephyr 3.2.0 .cpp and .c main function return types

### DISCUSSION THREAD

**Discussion by RadhikaMaheshwari-eaton** [zephyrproject-rtos/zephyr#53700] [Q&A]
Zephyr 3.2.0 is expecting .cpp file's main function to return int. While it is not throwing any error for void main in .c files, why is it so?
What should be the right approach to fix this issue?

Shouldn't we have same rules for both .cpp and .c files?

**Comment by henrikbrixandersen** [comment]
Please see https://github.com/zephyrproject-rtos/zephyr/pull/54628
